core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -248,6 +247,8 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, val brokerState
 
     initializeControllerContext()
 
+    val pendingPreferredReplicaElections = fetchPendingPreferredReplicaElections()
+    val (topicsToBeDeleted, topicsIneligibleForDeletion) = fetchTopicDeletionsInProgress(pendingPreferredReplicaElections)
```

Comment:
Good point. One way to do that is to de-couple the initialization of the state in TopicDeletionManager from the handling of topic deletion. So, the sequence will be 
(1) TopicDeletionManager.init(topicsToBeDeleted, topicsIneligibleForDeletion) // don't call resumeDeletions.
(2) replicaStateMachine.startup()
(3) partitionStateMachine.startup()
(4) maybeTriggerPartitionReassignment()
(5) topicDeletionManager.resumeDeletions().